client/app/services/axios.js

```diff
@@ -0,0 +1,24 @@
+import axiosLib from "axios";
+
+export const axios = axiosLib.create();
+
+const getData = ({ data }) => data;
+const getResponse = ({ response }) => Promise.reject(response);
```

Comment:
Is this needed to get a response object instead of error when request fails? If yes - I don't think it's so good idea. Angular's $http had similar behavior, and that's why we needed that PromiseRejectionError - to convert response to a Error instance, because in terms of error handling it's much more convenient if all promises are rejected with Error objects - global error handler does not need to perform additional checks, and if you handle them locally in component - you alread know what type of error you have and can, say, get `error.response` there.

Reply:
I mostly returned the response like this because it was the behavior $http had and I wanted to minimize the risks -- I suppose we hardly ever test errors in the application (unlike the success flow), it's a point to focus for future and add tests to alternative/error paths as well.

Regarding the global promise handler, I agree, it's easy to make sure it's indeed an error and show an error page in the end when the response is an Error instance. I think we can try to remove it when handling errors globally (not sure if you're doing this in #4525, I still need to sync myself to everything that's in there again).

In brief, I believe this is correct way to do it, but my fear is that changing this may cause issues like unhandled promises showing errors where you didn't want them and error notifications not showing what they should.

Reply:
Got it :+1: Let's merge it, and revisit it later if needed.